src/InterpreterIndex.cpp

```diff
@@ -241,10 +249,20 @@ class GenericIndex : public InterpreterIndex {
             return c;
         }
 
+        int reload(TupleRef* out, int max) override {
+            int c = 0;
+            max = std::max(max, Stream::BUFFER_SIZE);
```

Comment:
Did you mean std::min here?

Reply:
I did, yes.

Reply:
It should only be called by the Stream that it's coupled with, which already knows the limit, but there's nothing to enforce that so I check the size. At least, I do now that it uses std::min.